File: app/src/main/java/com/example/android/activitylifecyleexample/MainActivity.java
package com.example.android.activitylifecyleexample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    int numOfButtonClicks = 0;
    Button button;
    private static final String KEY_NUMCLICKS = "numClicks";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if(savedInstanceState != null) {
            numOfButtonClicks = savedInstanceState.getInt(KEY_NUMCLICKS);
        }

        button = (Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                numOfButtonClicks++;
                setButtonText();
            }
        });

        setButtonText();
    }

    private void setButtonText(){
        button.setText("Clicks: "+numOfButtonClicks);
    }

    @Override
    public void onSaveInstanceState(Bundle savedInstanceState) {
        super.onSaveInstanceState(savedInstanceState);
        savedInstanceState.putInt(KEY_NUMCLICKS, numOfButtonClicks);
    }
}

File: app/src/main/java/com/example/android/activitylifecyleexample/MainActivity.kt
package com.example.android.activitylifecyleexample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageButton
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    var numOfCupcakesBought = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (savedInstanceState != null)
            numOfCupcakesBought = savedInstanceState.getInt(NUM_OF_CUPCAKES)

        val cupcakeButton: ImageButton = findViewById(R.id.dessert_button)
        cupcakeButton.setOnClickListener {
            numOfCupcakesBought++;
            updateCupcakes()
        }

        updateCupcakes()
    }

    private fun updateCupcakes() {
        val amtTextView : TextView = findViewById(R.id.amount_sold_text)
        amtTextView.text = "$numOfCupcakesBought"
        val totalTextView : TextView = findViewById(R.id.revenue_text)
        totalTextView.text = "$${numOfCupcakesBought * 3}"

    }

    override fun onSaveInstanceState(savedInstanceState: Bundle) {
        super.onSaveInstanceState(savedInstanceState)
        savedInstanceState.putInt(NUM_OF_CUPCAKES, numOfCupcakesBought)
    }

    companion object {
        const val NUM_OF_CUPCAKES = "numOfCupcakes"
    }

}